spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestDataSourceOptions.java

```diff
@@ -448,4 +448,37 @@ public void testExtraSnapshotMetadataWithSQL() throws InterruptedException, IOEx
     Assert.assertTrue(threadNames.contains(null));
     Assert.assertTrue(threadNames.contains("test-extra-commit-message-writer-thread"));
   }
+
+  @Test
+  public void testExtraSnapshotMetadataForSqlInPySpark() throws IOException {
+    String tableLocation = temp.newFolder("iceberg-table").toString();
+    HadoopTables tables = new HadoopTables(CONF);
+
+    Table table =
+        tables.create(SCHEMA, PartitionSpec.unpartitioned(), Maps.newHashMap(), tableLocation);
+
+    List<SimpleRecord> expectedRecords =
+        Lists.newArrayList(new SimpleRecord(1, "a"), new SimpleRecord(2, "b"));
+    Dataset<Row> originalDf = spark.createDataFrame(expectedRecords, SimpleRecord.class);
+    originalDf.select("id", "data").write().format("iceberg").mode("append").save(tableLocation);
+    spark.read().format("iceberg").load(tableLocation).createOrReplaceTempView("target");
+    spark.conf().set("spark.snapshot-property.extra-key", "someValue");
```

Comment:
We typically use `spark.iceberg` for Iceberg specific things. What do you think about using that here instead of just `spark`?

Reply:
ok it's better to have everything similar :)
thanks for the review